pkg/ddl/job_scheduler.go

```diff
@@ -325,31 +325,43 @@ func (s *jobScheduler) schedule() error {
 		case <-s.schCtx.Done():
 			return s.schCtx.Err()
 		}
-		if err := s.checkAndUpdateClusterState(false); err != nil {
-			continue
-		}
+
 		failpoint.InjectCall("beforeLoadAndDeliverJobs")
 		if err := s.loadAndDeliverJobs(se); err != nil {
 			logutil.SampleLogger().Warn("load and deliver jobs failed", zap.Error(err))
 		}
 	}
 }
 
-// TODO make it run in a separate routine.
-func (s *jobScheduler) checkAndUpdateClusterState(needUpdate bool) error {
-	select {
-	case _, ok := <-s.serverStateSyncer.WatchChan():
-		if !ok {
-			// TODO serverStateSyncer should only be started when we are the owner, and use
-			// the context of scheduler, will refactor it later.
-			s.serverStateSyncer.Rewatch(s.ddlCtx.ctx)
+// updateClusterStateLoop updates the cluster state in loop.
+func (s *jobScheduler) updateClusterStateLoop() {
+	var err error = nil
+
+	defer tidbutil.Recover(metrics.LabelDDL, "updateClusterStateLoop", nil, true)
+	// server stateSyncer uses the scheduler context, so rewatch again with the scheduler context.
+	s.serverStateSyncer.Rewatch(s.schCtx)
+
+	for {
+		err = s.checkAndUpdateClusterState()
+		if err != nil {
+			logutil.DDLLogger().Warn("failed to check and update cluster state", zap.Error(err))
 		}
-	default:
-		if !needUpdate {
-			return nil
+
+		select {
```

Comment:
we better add a ticker based check in case checkAndUpdateClusterState fail and there are no further change to the key, or like what `syncJobSchemaVer` does, to make sure we successfully processed existing value and incremental changes.

Reply:
> we better add a ticker based check in case checkAndUpdateClusterState fail and there are no further change to the key, or like what `syncJobSchemaVer` does, to make sure we successfully processed existing value and incremental changes.

done